src/scatter_buildblock/ScatterSimulation.cxx

```diff
@@ -826,18 +832,52 @@ ScatterSimulation::downsample_scanner(int new_num_rings, int new_num_dets)
         if(downsample_scanner_dets > 0)
             new_num_dets = downsample_scanner_dets;
         else
-            return Succeeded::no;
+            new_num_dets=proj_data_info_cyl_noarc_cor_sptr->get_scanner_ptr()->get_num_detectors_per_ring();
     }
 
     const Scanner *const old_scanner_ptr = this->proj_data_info_cyl_noarc_cor_sptr->get_scanner_ptr();
     shared_ptr<Scanner> new_scanner_sptr( new Scanner(*old_scanner_ptr));
-
-    // preserve the length of the scanner
-    float scanner_length = new_scanner_sptr->get_num_rings()* new_scanner_sptr->get_ring_spacing();
+    // preserve the length of the scanner the following includes gaps
+    float scanner_length = new_scanner_sptr->get_num_axial_buckets()*
```

Comment:
the problem with the test is here. Cylindrical geometry has axial_block_spacing set to -1. I guess we should have a different length calculation for Cylindrical?

Reply:
test runs OK with this modification